test/index.js

```diff
@@ -1463,6 +1463,139 @@ describe('Nightmare', function () {
       didSubscribe.should.be.false;
     });
 
+    it('should subscribe to frames when requested necessary', function(done) {
+      var didSubscribe = false;
+      var didUnsubscribe = false;
+      var FrameManager = require('../lib/frame-manager.js');
+      var fn;
+      var manager = FrameManager({
+        webContents: {
+          debugger: {
+            isAttached: function() { return true; },
+            sendCommand: function(command) { if (command === 'DOM.highlightRect') { fn('mock-data'); }}
+          },
+          beginFrameSubscription: function(_fn) { didSubscribe = true; fn = _fn; },
+          endFrameSubscription: function() { didUnsubscribe = true; },
+          executeJavaScript: function() {}
+        }
+      });
+      manager.requestFrame(function (data) {
+        didSubscribe.should.be.true;
+        didUnsubscribe.should.be.true;
+        data.should.equal('mock-data');
+        done();
+      });
+    });
+
+    it('should support multiple concurrent frame subscriptions', function(done) {
+      var subscribeCount = 0;
+      var unsubscribeCount = 0;
+      var FrameManager = require('../lib/frame-manager.js');
+      var fn = null;
+      var assert = require('assert');
+      var async = require('async');
+      var manager = FrameManager({
+        webContents: {
+          debugger: {
+            isAttached: function() { return true; },
+            sendCommand: function(command) {
+              if (command === 'DOM.highlightRect') {
+                setTimeout(function () {
+                  fn('mock-data');
+                }, 100);
+              }
+            }
+          },
+          beginFrameSubscription: function(_fn) { subscribeCount += 1; assert.strictEqual(fn, null); fn = _fn; },
```

Comment:
Hmm, I would say it's on the fence. On one hand, it's faking out the `beginFrameSubscription` to prevent it's `receiveFrame` multiple times (i.e. programmatic). On the other hand, it's verifying that our `async.times` doesn't invoke the `beginFrameSubscription` multiple times (i.e. assertion)